lib/foreman_tasks/tasks/test.rake

```diff
@@ -1,14 +1,15 @@
 namespace :test do
-  task :foreman_tasks => 'db:test:prepare' do
-    test_task = Rake::TestTask.new('foreman_tasks_test_task') do |t|
-      t.libs << ['test', "#{ForemanTasks::Engine.root}/test"]
-      t.test_files = Dir["#{ForemanTasks::Engine.root}/test/**/*_test.rb"]
-      t.verbose = true
-      t.warning = false
-    end
-
-    Rake::Task[test_task.name].invoke
+  desc 'Foreman Tasks plugin tests'
+  Rake::TestTask.new(foreman_tasks: 'db:test:prepare') do |t|
+    t.libs << ForemanTasks::Engine.root.join('test')
+    t.pattern = ForemanTasks::Engine.root.join('test/**/*_test.rb')
+    t.test_files = [Rails.root.join('test/unit/foreman/access_permissions_test.rb')]
```

Comment:
In OpenSCAP I saw a dedicated task that ran the tests from core where it modified things. Especially when some controllers are extended I think that's a good pattern to keep, and it may help with setting up correct load paths. I'm now debating that approach as well.